### ISSUE TITLE
Flatpak build

### ISSUE THREAD

**Issue by LeandroStanger** [ventoy/Ventoy#111]
Please, consider supporting [Flatpak](http://docs.flatpak.org/en/latest/introduction.html#reasons-to-use-flatpak) and publishing Ventoy on [Flathub](https://github.com/flathub/flathub/wiki/App-Submission).

**Comment by blackcrack** [comment]
You stand and want ...
why do you not the work.. or other Peoples, i guess, 
Ventoy have a lot work on other more important things..

**Comment by blackcrack** [comment]
if you it want, then do it self.. Flatpack is for linux, this is nice.. but make more work for Ventoy Group 
and i guess, they are more WinNt users, the Linuxbinary is from a single entusiast or so.. well, i guess not to support flatpack
https://github.com/flathub/flathub/wiki/App-Submission
>Who can use Flathub
>
>Flathub is primarily intended for use by developers who want to distribute desktop apps on Linux. >Apps must either be redistributable or be available as a third party download.
>
>If you are responsible for a proprietary app that you would like to distribute through Flathub, we’d >love to speak to you, and would encourage you to contact the Flathub admins.

you can close this issue, i don't see a land..

**Comment by monterrr** [comment]
I'm with this, please publish a flatpak on flathub.org

**Comment by rugk** [comment]
It would be nice if you could publish this as a [flatpak](https://flatpak.org/) on [flathub](https://flathub.org/) e.g.
Flatpaks are a new software distribution mechanism for Linux distros, can thus installed on any distro and are easy to update. 
Here is [how to get started](http://docs.flatpak.org/en/latest/getting-started.html). It makes maintainance actually easier instead of having to do custom builds for each distro.

For reference, DVD stuff seems to work in Flatpak, e.g. there are these tools:
* https://flathub.org/apps/details/com.poweriso.PowerISO
* https://flathub.org/apps/details/com.makemkv.MakeMKV

**Comment by BeatLink** [comment]
I guess with ventoy now having an official native GUI, implementing a flatpak would be much easier?

**Comment by Arcitec** [comment]
> I guess with ventoy now having an official native GUI, implementing a flatpak would be much easier?

Yes it would be very easy now. Simply make a flatpak definition that contains the appropriate runtime environment, downloads the latest source code, compiles Ventoy, and refers to the correct GUI executable. It would need permissions to read/write the whole Host filesystem with `--device=all`. That's it, I think...

**Comment by iMonZ** [comment]
So any news on that?
Is the Ventoy Project dead?
Since this is really important for linux

**Comment by BeatLink** [comment]
The project is very much alive. From what I can see, a commit was made only 4 days ago. I guess the maintainers are looking for someone to volunteer to implement this. I would be willing to but i have too many engagements at the moment to find any free time.

**Comment by tim77** [comment]
Ventoy requires root privileges to work. How you offer to package it via Flatpak?

**Comment by Pentaphon** [comment]
> I guess with ventoy now having an official native GUI, implementing a flatpak would be much easier?

I agree. This should be on flatpak so that everybody can install it easily for Linux.

**Comment by tim77** [comment]
> flatpak applications do have the capability to access storage devices. I presume that would be where the need for root privileges would originate?

> For example, --filesystem=host and --device=all could be used here.

No unfortunately. Problem with Ventoy not because of FS permissions and accessing devices. But it could run i suppose with `flatpak-spawn` which is far from optimal usage and what about flatpak in general.

**Comment by BeatLink** [comment]
If root permissions is the issue, couldnt the GUI be launched with root permissions using gksu or PAM in the desktop file exec line for authentication?

**Comment by Mr-Bajs** [comment]
Would be very nice if andFlakpak/Flathub option where available, if that would be possible.

**Comment by Pentaphon** [comment]
> Would be very nice if andFlakpak/Flathub option where available, if that would be possible.

If you want it, give this thread a thumbs up add to the number of people who want this.

**Comment by flavio20002** [comment]
Plase, do it. Ventoy it's an amazing app, and Flatpak is the new standard!

**Comment by mm12** [comment]
see also: 
#1634 #702

**Comment by NaniNoni** [comment]
Any progress?

**Comment by LorenzoIanotto** [comment]
I am working on it (https://github.com/LorenzoIanotto/VentoyUDisks.git).
I have managed to compile a WebUI for Linux which doesn't need root privileges using UDisks2, but it is buggy and segfaults all the time (and probably it is not that memory-safe...). Help would be appreciated.

**Comment by boustanihani** [comment]
Hope we get this soon :)

**Comment by MurzNN** [comment]
While waiting for Flatpak, found the AppImage version here https://github.com/ryuuzaki42/Ventoy_AppImage - at least, better than nothing ;)

**Comment by 54145a** [comment]
Any progress on this?

**Comment by shvchk** [comment]
@MurzNN also, there's a Nix package: https://search.nixos.org/packages?query=ventoy
Nix package manager works on ~any Linux distro, it's basically just `sudo apt install nix` or the equivalent on other distros.

More info: https://wiki.archlinux.org/title/Nix (more or less the same for any distro)

**Comment by bojogon7** [comment]
just saying you can use [bootqt](https://flathub.org/apps/io.github.giantpinkrobots.bootqt) from flathub to make ventoy USB, for anyone interested
